django_toolkit/concurrent/locks.py

```diff
@@ -0,0 +1,48 @@
+from django.core.cache import caches
+
+
+class LockActiveError(Exception):
+    pass
+
+
+class BaseLock(object):
+
+    def __init__(self):
+        self.active = False
+
+
+class LocalMemoryLock(BaseLock):
+    """
+    A context manager to handle a lock status in memory
+
+    The active status is True on __enter__ and False on __exit__
+    """
+
+    def __enter__(self):
+        self.active = True
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.active = False
+
+
+class CacheLock(BaseLock):
+
+    cache = caches['locks']
+
+    def __init__(self, key, expire=0):
+        super(CacheLock, self).__init__()
+        self._key = key
+        self._expire = expire
+
+    def __enter__(self):
+        self.active = self.cache.add(self._key, True, self._expire)
+
+        if not self.active:
+            raise LockActiveError('For key {key}'.format(key=self._key))
```

Comment:
The user of this context manager would have add a `try... except` block to catch `LockActiveError` and finish the task when it happens, right?